api/src/utils/auth-utils.ts

```diff
@@ -18,7 +18,8 @@ const hasRole = (
 ): boolean =>
   user &&
   roles.includes(user.role) &&
-  (user.hasRoleApproved || !requireApproved);
+  (user.onboardingStatus === onboardingStatusEnum.ONBOARDED ||
```

Comment:
May want to replace this condition with the `isOnboarded` function you added below.
```suggestion
  (isOnboarded(user) ||
```
You'll have to move this function lower in the file if you want to do that though, so it's really not 100% necessary.